interface/forms/fee_sheet/review/js/fee_sheet_core.js

```diff
@@ -117,21 +117,37 @@ function parse_row_justify(row)
     return retval;
 }
 
-function justify_start(evt)
-{
-    var jqElem=$(this);
-    var parent=jqElem.parent()
-    var template_div=parent.find("div.justify_template");
-    if(template_div.length==0)
-    {
-        var template_div=$("<div class='justify_template'></div>");
-        template_div.attr("data-bind","template: {name: 'justify-display', data: justify}");
-        jqElem.after(template_div);
-    }
-    $(".cancel_dialog").click();
-    var current_justify_choices=parse_row_justify(parent.parent());
-    var justify_model=new fee_sheet_justify_view_model(parent.attr("billing_id"),enc,pid,current_justify_choices);
-    ko.applyBindings(justify_model,template_div.get(0));
+function justify_start(evt) {
+    const jqElem = $(this); // make sure is in scope after save.
+    let myForm = document.getElementById('fee_sheet_form');
+    let formData = new FormData(myForm);
+    formData.append('running_as_ajax', "1");
+    formData.append('dx_update', "1");
+    // save current form
+    $.ajax({
+        url: fee_sheet_new,
+        processData: false,
+        contentType: false,
+        cache: false,
+        type: 'POST',
+        data: formData,
+        beforeSend: function () {
+            top.restoreSession();
+        }
+    }).done(function (data) {
```

Comment:
not finding .done is .success what is used now?
https://api.jquery.com/jQuery.ajax/

Reply:
done,fail and always are promise fulfilments. 
success can be used inside the ajax request but is deprecated anywhere other.
Could have written this as
```suggestion
  $.ajax({
        url: fee_sheet_new,
        processData: false,
        contentType: false,
        cache: false,
        type: 'POST',
        data: formData,
        beforeSend: function () {
            top.restoreSession();
        },
        success: function (data) {
            var parent=jqElem.parent()
            var template_div=parent.find("div.justify_template");
            if(template_div.length==0)
            {
                var template_div=$("<div class='justify_template'></div>");
                template_div.attr("data-bind","template: {name: 'justify-display', data: justify}");
                jqElem.after(template_div);
            }
            $(".cancel_dialog").click();
            var current_justify_choices=parse_row_justify(parent.parent());
            var justify_model=new fee_sheet_justify_view_model(parent.attr("billing_id"),enc,pid,current_justify_choices);
            ko.applyBindings(justify_model,template_div.get(0));
        }
    });
```
but not
```suggestion
   ajax().success( function (data) {
            var parent=jqElem.parent()
            var template_div=parent.find("div.justify_template");
            if(template_div.length==0)
            {
                var template_div=$("<div class='justify_template'></div>");
                template_div.attr("data-bind","template: {name: 'justify-display', data: justify}");
                jqElem.after(template_div);
            }
            $(".cancel_dialog").click();
            var current_justify_choices=parse_row_justify(parent.parent());
            var justify_model=new fee_sheet_justify_view_model(parent.attr("billing_id"),enc,pid,current_justify_choices);
            ko.applyBindings(justify_model,template_div.get(0));
        })
```
Preface would have been using a fetch post promise but i'm still a little apprehensive with replacing current code with es6 pattern.
I'd highly recm if trying to improve JS skills to look over using the different variations of fetch promise and async/await. Still, ajax as promises is a very useful and promising pattern as well. (see what I did there?...promising)

Reply:
:) ok

Reply:
ya want to take a shot at writing it as a fetch post?

Reply:
:hear_no_evil:

Reply:
could it be as simple as removing the .ajax and replacing with the below plus switching the done method to then?
```
fetch('../../fee_sheet_new.php', {
        method: 'POST',
        body: formData

    }).then(function (data) {
````